lumicks/pylake/detail/utilities.py

```diff
@@ -44,3 +45,12 @@ def find_contiguous(mask):
     ranges = np.argwhere(change_points == 1).reshape(-1, 2)
     run_lengths = np.diff(ranges, axis=1).squeeze()
     return ranges, np.atleast_1d(run_lengths)
+
+
+def downsample(data, factor, reduce):
```

Comment:
It would be good to give it a docstring and tests at this point. Previously, it was used as a private function in the same file so the usage was the main thing defining it. Now that it's become (internally) reusable, it needs a docstring and a short unit test. (Internal API users are very similar to public API users.)

Reply:
agreed, but I do like that it's separated out now

Reply:
Yes, will add a docstring :)